metadata-ingestion/src/datahub/ingestion/source/file.py

```diff
@@ -236,6 +248,55 @@ def get_workunits_internal(
                 else:
                     yield MetadataWorkUnit(id, mce=obj)
 
+    def _convert_usage_aggregation_class(
+        self, obj: UsageAggregationClass
+    ) -> Optional[MetadataChangeProposalWrapper]:
+        # Legacy usage aggregation only supported dataset usage stats
+        if guess_entity_type(obj.resource) == DatasetUrn.ENTITY_TYPE:
```

Comment:
can we move this conversion into a separate method/file?

ideally we should add a unit test for this logic one it's moved